src/components/Overlay/provider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+import { RouteComponentProps, withRouter } from "react-router";
+import {
+  InnerOverlayContextInterface,
+  OverlayContext,
+  OverlayContextInterface,
+  OverlayTheme,
+  OverlayType
+} from "./context";
+
+class Provider extends React.Component<
+  RouteComponentProps<{}>,
+  OverlayContextInterface
+> {
+  notificationCloseDelay = 2500;
+  constructor(props) {
```

Comment:
You don't need to use the constructor here, class prop also works:
```
state: OverlayContextInterface = {
...
}
```

Reply:
But methods don't.